pennylane/_qubit_device.py

```diff
@@ -279,7 +279,18 @@ def execute(self, circuit, **kwargs):
         # generate computational basis samples
         sample_type = (SampleMP, CountsMP, ClassicalShadowMP, ShadowExpvalMP)
         if self.shots is not None or any(isinstance(m, sample_type) for m in circuit.measurements):
+            # Lightning does not support apply(rotations) anymore, so we need to rotate here
+            if hasattr(self, "name") and isinstance(self.name, str) and "Lightning" in self.name:
```

Comment:
This is quite defensive. Is it not guaranteed that all `QubitDevice` subclasses have a string `name` attribute? 

Furthermore, should we:
1. Make the check `"Lightning" in self.name` case-insensitive?
2. Match the known device names directly?
3. Store this result in a variable so we don't have to repeat the same check below?

Reply:
1. I don't think that's necessary.
2. The `name` attribute is quite verbose, and `short_name` may not be defined (I think), so checking for "Lightning" is specific enough at this point, I think. Note that all this can be removed when we move the Lightning plugins to the new API (likely next release).
3. Good point, will do.

Reply:
> This is quite defensive. Is it not guaranteed that all QubitDevice subclasses have a string name attribute?

Sorry, forgot to comment on that. You're right, but it seems some devices have name `None` so I'd rather err on the side of caution.